content/doc/book/using/using-agents.adoc

```diff
@@ -0,0 +1,195 @@
+
+== Configuring agents with Docker
+
+Jenkins agents can be launched in physical machines, virtual machines, Kubernetes clusters, and with Docker images.
+This section connects Docker agents to Jenkins with SSH.
+
+=== Environment
+
+To run this guide you will need a machine with:
+
+* Java installation
+* Jenkins installation
+* Docker installation
+* SSH key pair
+
+[NOTE]
+====
+If you need help to install Java, Jenkins and Docker please visit the section link:/doc/book/installing/docker/[Installing Jenkins.]
+====
+
+=== Generating an SSH key pair
+
+1. In a terminal window run the command: `ssh-keygen -f ~/.ssh/jenkins_agent_key`
+2. Provide a passphrase to use with the key
+3. Confirm the output looks something like this:
++
+[source,bash]
+----
+ubuntu@desktop:~$ ssh-keygen -f ~/.ssh/jenkins_agent_key
+Generating public/private rsa key pair.
+Enter passphrase (empty for no passphrase):
+Enter same passphrase again:
+Your identification has been saved in /home/ubuntu/.ssh/jenkins_agent_key
+Your public key has been saved in /home/ubuntu/.ssh/jenkins_agent_key.pub
+The key fingerprint is:
+SHA256:XqxxjqsLlvDD0ZHm9Y2iR7zC6IbsUlMEHo3ffy8TzGs
+The key's randomart image is:
++---[RSA 3072]----+
+|  o+             |
+| ...o  .         |
+|  .o .+ .        |
+|    o+.+ o o     |
+|  ... o.So* .    |
+|  o+ = +.X=      |
+| o oO + *..+     |
+|. oo.o o .E .    |
+| o... oo.. o     |
++----[SHA256]-----+
+----
+
+==== Create a Jenkins SSH credential
+
+1. Go to your Jenkins dashboard;
+2. Go to `Manage Jenkins` option in main menu and click on credentials button;
++
+image:node/credentials-1.png[credentials menu,700]
+
+3. select the drop option `Add Credentials` from the global item;
++
+image:node/credentials-2.png[add credentials option,700]
+
+4. Fill the form:
+* Kind:  SSH Username with private key;
+* id: jenkins
+* description: The jenkins ssh key
+* username: jenkins
+* Private Key: select `Enter directly` and press the Add button to insert your private key from `~/.ssh/jenkins_agent_key`
+* Passphrase: fill your passphrase used to generate the SSH key pair and then press OK
+    image:node/credentials-3.png[credentials filled form,700]
+
+=== Creating your Docker agent
+
+Here we will use the link:https://github.com/jenkinsci/docker-ssh-agent[docker-ssh-agent image] to create the agent containers.
+
+1. run the command to start your first agent:
++
+[source,bash]
+----
+docker run -d --rm --name=agent1 -p 22:22 \
+-e "JENKINS_AGENT_SSH_PUBKEY=[your-public-key]" \
+jenkins/ssh-agent:alpine
+----
++
+[NOTE]
+====
+* Remember to replace the tag [your-public-key] for your own SSH *public* key.
+* Your public key in this example is: `cat ~/.ssh/jenkins_agent_key.pub`
+====
+2. Now run the following command to update the container environment:
++
+[source,bash]
+----
+docker exec  agent1 sh -c "env | egrep -v "^(HOME=|USER=|MAIL=|LC_ALL=|LS_COLORS=|LANG=|HOSTNAME=|PWD=|TERM=|SHLVL=|LANGUAGE=|_=)" >> /etc/environment"
+----
```

Comment:
Narrow the example source code so that it does not disrupt screen layout for the entire page.
```suggestion
[source,bash]
----
$ VARS1="HOME=|USER=|MAIL=|LC_ALL=|LS_COLORS=|LANG="
$ VARS2="HOSTNAME=|PWD=|TERM=|SHLVL=|LANGUAGE=|_="
$ VARS="${VARS1}|${VARS2}"
$ docker exec agent1 sh -c "env | egrep -v "^(${VARS})" >> /etc/environment"
----
```